lib/motorcontrollers/controller.cpp

```diff
@@ -0,0 +1,130 @@
+#include "controller.hpp"
+#include <vector>
+
+namespace hyped::motorcontrollers {
+
+  
+
+Controller::Controller(core::ILogger &logger) : logger_(logger))
+{
+  //TODO: Implement Enum for warning codes
+  // Implement function to iterate through warning codes given a warning bit selection
+  // and output a vector of enums
+  enum warnings {ControllerTemperatureExceeded, MotorTemperatureExceeded, DCLinkUnderVoltage, 
```

Comment:
I would consider enums inside functions to be bad practice.

Reply:
I will get that changed asap